code/game/machinery/atmoalter/canister.dm

```diff
@@ -366,72 +366,72 @@ update_flag
 		return STATUS_CLOSE
 	return ..()
 
-/obj/machinery/portable_atmospherics/canister/phoron/New()
+/obj/machinery/portable_atmospherics/canister/phoron/New(var/atom/location, var/direction, var/nocircuit = FALSE)
 	..()
 
 	src.air_contents.adjust_gas(MATERIAL_PHORON, MolesForPressure())
 	src.update_icon()
 	return 1
 
-/obj/machinery/portable_atmospherics/canister/oxygen/New()
+/obj/machinery/portable_atmospherics/canister/oxygen/New(var/atom/location, var/direction, var/nocircuit = FALSE)
 	..()
 
 	src.air_contents.adjust_gas("oxygen", MolesForPressure())
 	src.update_icon()
 	return 1
 
-/obj/machinery/portable_atmospherics/canister/hydrogen/New()
+/obj/machinery/portable_atmospherics/canister/hydrogen/New(var/atom/location, var/direction, var/nocircuit = FALSE)
 	..()
 	src.air_contents.adjust_gas("hydrogen", MolesForPressure())
 	src.update_icon()
 	return 1
 
-/obj/machinery/portable_atmospherics/canister/oxygen/prechilled/New()
+/obj/machinery/portable_atmospherics/canister/oxygen/prechilled/New(var/atom/location, var/direction, var/nocircuit = FALSE)
 	..()
 	src.air_contents.temperature = 80
 	src.update_icon()
 	return 1
 
-/obj/machinery/portable_atmospherics/canister/sleeping_agent/New()
+/obj/machinery/portable_atmospherics/canister/sleeping_agent/New(var/atom/location, var/direction, var/nocircuit = FALSE)
 	..()
 
 	air_contents.adjust_gas("sleeping_agent", MolesForPressure())
 	src.update_icon()
 	return 1
 
 //Dirty way to fill room with gas. However it is a bit easier to do than creating some floor/engine/n2o -rastaf0
-/obj/machinery/portable_atmospherics/canister/sleeping_agent/roomfiller/New()
+/obj/machinery/portable_atmospherics/canister/sleeping_agent/roomfiller/New(var/atom/location, var/direction, var/nocircuit = FALSE)
 	..()
 	air_contents.gas["sleeping_agent"] = 9*4000
 	spawn(10)
-		var/turf/simulated/location = src.loc
+		var/turf/simulated/location2 = src.loc
```

Comment:
whats this change for?

Reply:
because of this:
`code\game\machinery\atmoalter\canister.dm:407:error: location: duplicate definition
code\game\machinery\atmoalter\canister.dm:403:error: location: previous definition`
and if you try to remove var/turf/simulated/location2 = src.loc leaving location.air and location.assume_air it will complain about undefiend var:
`location.air: undefined var`
I forgot to change second `location`, pushed commit that fixes it already